ambry-coordinator/src/test/java/com.github.ambry.coordinator/MockDataNode.java

```diff
@@ -35,36 +32,16 @@
 public class MockDataNode {
   private DataNodeId dataNodeId;
 
-  private Map<BlobId, MaterializedBlob> blobs;
```

Comment:
Siva, for the change in this PR, you should not have to worry about `ByteBufferInputStream` and things like that. This is what I mean by complicating. My claim is that `MockDataNode` will need zero changes.

One thing to keep in mind is that we don't want to make any major changes to the coordinator package - only the minimal amount of changes as required - which means anything that can be avoided should be avoided. We do not want to optimize code that will soon be removed.